src/VisualStudio/IntegrationTest/New.IntegrationTests/AbstractIntegrationTest.cs

```diff
@@ -23,8 +58,16 @@ public override async Task InitializeAsync()
         {
             await base.InitializeAsync();
 
+            s_listenerProvider ??= await TestServices.Shell.GetComponentModelServiceAsync<AsynchronousOperationListenerProvider>(HangMitigatingCancellationToken);
+
             if (await TestServices.SolutionExplorer.IsSolutionOpenAsync(HangMitigatingCancellationToken))
             {
+                var dte = await TestServices.Shell.GetRequiredGlobalServiceAsync<SDTE, EnvDTE.DTE>(HangMitigatingCancellationToken);
+                if (dte.Debugger.CurrentMode != EnvDTE.dbgDebugMode.dbgDesignMode)
+                {
+                    dte.Debugger.TerminateAll();
```

Comment:
If a test fails during a debugging session, it will not stop the debugger before exiting the test method. This method cleans up from whatever state the test completed in so the failure is isolated to just the test with the assertion failure.